File: src/main/java/co/herod/adbwrapper/rx/FixedDurationTransformer.java
package co.herod.adbwrapper.rx;

import io.reactivex.Observable;
import io.reactivex.ObservableTransformer;
import org.jetbrains.annotations.NotNull;

import java.util.concurrent.TimeUnit;

public class FixedDurationTransformer implements ObservableTransformer<String, String> {

    private final int timeout;
    private final TimeUnit timeUnit;

    public FixedDurationTransformer(int timeout, TimeUnit timeUnit) {

        this.timeout = timeout;
        this.timeUnit = timeUnit;
    }

    @Override
    public Observable<String> apply(@NotNull Observable<String> upstream) {
        return upstream.retry().timeout(timeout, timeUnit);
    }
}

File: src/main/java/co/herod/adbwrapper/rx/FixedDurationTransformer.kt
package co.herod.adbwrapper.rx

import io.reactivex.Observable
import io.reactivex.ObservableTransformer

import java.util.concurrent.TimeUnit

open class FixedDurationTransformer(
        private val timeout: Int,
        private val timeUnit: TimeUnit
) : ObservableTransformer<String, String> {

    override fun apply(upstream: Observable<String>): Observable<String> =
            upstream.retry().timeout(timeout.toLong(), timeUnit)
}
